aiobotocore/endpoint.py

```diff
@@ -51,43 +69,43 @@ def __dir__(self):
         return attrs
 
     def __del__(self):
-        self.close()
+        # callers may not read the Body resulting in an unclosed ClientResponse
+        if self._loop.is_running():
+            asyncio.ensure_future(self.__response.release(), loop=self._loop)
+        else:
+            self.close()
 
     if PY_35:
         @asyncio.coroutine
         def __aenter__(self):
-            return self.__response.__aenter__()
+            yield from self.__response.__aenter__()
+            return self
 
         @asyncio.coroutine
         def __aexit__(self, exc_type, exc_val, exc_tb):
-            return self.__response.__aexit__(exc_type, exc_val, exc_tb)
+            return (yield from self.__response.__aexit__(exc_type,
+                                                         exc_val, exc_tb))
 
     def close(self):
         self.__response.close()
 
 
-class ClientResponseProxy:
+class ClientResponseProxy(ClientResponse):
     """Proxy object for http response useful for porting from
     botocore underlying http library."""
 
     def __init__(self, *args, **kwargs):
-        self._impl = ClientResponse(*args, **kwargs)
-        self._body = None
+        super().__init__(*args, **kwargs)
 
     def __getattr__(self, item):
         if item == 'status_code':
-            return getattr(self._impl, 'status')
+            return self.status
```

Comment:
Better just drop `__getattr__`, that was my stupid idea and laziness )) adding required properties and methods would be much nicer and more explicit.